internal/civisibility/utils/net/client_test.go

```diff
@@ -0,0 +1,231 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024 Datadog, Inc.
+
+package net
+
+import (
+	"os"
+	"reflect"
+	"strings"
+	"testing"
+)
+
+func saveEnv() []string {
+	return os.Environ()
+}
+
+func restoreEnv(env []string) {
+	os.Clearenv()
+	for _, e := range env {
+		kv := strings.SplitN(e, "=", 2)
+		os.Setenv(kv[0], kv[1])
+	}
+}
+
+func TestNewClient_DefaultValues(t *testing.T) {
+	origEnv := saveEnv()
+	path := os.Getenv("PATH")
+	defer restoreEnv(origEnv)
+
+	os.Clearenv()
+	os.Setenv("PATH", path)
```

Comment:
I know but the implementation is different though, `t.setEnv` sets the environment variable and restore the old value. In our case we are deleting all environment variables and setting new ones (To avoid any flakiness due CI env-vars).